### ISSUE TITLE
NODEJS Compiler Issue Version 48 vs Version 51

### ISSUE THREAD

**Issue by erfind** [nfarina/homebridge#1500]
I'm on a Raspberry Pi, and have run into this issue multiple times, usually resulting in nuking the entire Raspberry Pi as starting over again.  It happens whenever I try to update to the newest version of homebridge using

sudo npm -g update

then like clockwork this crap starts... :( any ideas on what I can do to remedy this?

Sep 19 11:07:46 OfficeHomebridge systemd[1]: Starting Node.js HomeKit Server...
Sep 19 11:07:46 OfficeHomebridge systemd[1]: Started Node.js HomeKit Server.
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: *** WARNING *** The program 'nodejs' uses the Apple Bonjour compatibility layer of Avahi.
Sep 19 11:07:47 OfficeHomebridge nodejs[3682]: *** WARNING *** The program 'nodejs' uses the Apple Bonjour compatibility layer of Avahi.
Sep 19 11:07:47 OfficeHomebridge nodejs[3682]: *** WARNING *** Please fix your application to use the native API of Avahi!
Sep 19 11:07:47 OfficeHomebridge nodejs[3682]: *** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=nodejs>
Sep 19 11:07:47 OfficeHomebridge nodejs[3682]: *** WARNING *** The program 'nodejs' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
Sep 19 11:07:47 OfficeHomebridge nodejs[3682]: *** WARNING *** Please fix your application to use the native API of Avahi!
Sep 19 11:07:47 OfficeHomebridge nodejs[3682]: *** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=nodejs&f=DNSServiceRegister>
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: *** WARNING *** Please fix your application to use the native API of Avahi!
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: *** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=nodejs>
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: *** WARNING *** The program 'nodejs' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: *** WARNING *** Please fix your application to use the native API of Avahi!
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: *** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=nodejs&f=DNSServiceRegister>
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: /usr/lib/node_modules/hap-nodejs/node_modules/bindings/bindings.js:83
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: throw e
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: ^
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: Error: The module '/usr/lib/node_modules/hap-nodejs/node_modules/ed25519/build/Release/ed25519.node'
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: was compiled against a different Node.js version using
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: NODE_MODULE_VERSION 48. This version of Node.js requires
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: NODE_MODULE_VERSION 51. Please try re-compiling or re-installing
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: the module (for instance, using `npm rebuild` or`npm install`).
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: at Object.Module._extensions..node (module.js:598:18)
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: at Module.load (module.js:488:32)
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: at tryModuleLoad (module.js:447:12)
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: at Function.Module._load (module.js:439:3)
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: at Module.require (module.js:498:17)
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: at require (internal/module.js:20:19)
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: at bindings (/usr/lib/node_modules/hap-nodejs/node_modules/bindings/bindings.js:76:44)
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: at Object.<anonymous> (/usr/lib/node_modules/hap-nodejs/node_modules/ed25519/index.js:1:99)
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: at Module._compile (module.js:571:32)
Sep 19 11:07:47 OfficeHomebridge homebridge[3682]: at Object.Module._extensions..js (module.js:580:10)
Sep 19 11:07:47 OfficeHomebridge systemd[1]: homebridge.service: main process exited, code=exited, status=1/FAILURE
Sep 19 11:07:47 OfficeHomebridge systemd[1]: Unit homebridge.service entered failed state.
Sep 19 11:07:52 OfficeHomebridge systemd[1]: homebridge.service holdoff time over, scheduling restart.
Sep 19 11:07:52 OfficeHomebridge systemd[1]: Stopping Node.js HomeKit Server...

**Comment by ipankrub** [comment]
I experienced this before after I updated NodeJS to 7.6.0 as required by plugin (broadlink).

What I did was remove homebridge, plugins, homebridge.service and everything related to the homebridge setting-up guide. Then reinstall NodeJS again then follow the homebridge (homebridge installed the latest or it will need recompile/reinstall again). 

Hope this help, I'm also new to raspberry pi but this fix my problem.
